src/raster/r.suitability.regions/r.suitability.regions.py

```diff
@@ -0,0 +1,410 @@
+    tmpf = prefix + str(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
+    CLEAN_LAY.append(tmpf)
+    return tmpf
+
+
+def tmpmask(input, absmin):
+    """Create tmp mask"""
+    rules = "*:{}:1".format(absmin)
+    tmprecode = tmpname("tmprecode")
+    gs.write_command(
+        "r.recode", input=input, output=tmprecode, rule="-", stdin=rules, quiet=True
+    )
+    return tmprecode
+
+
+def main(options, flags):
+
+    suitrast = options["suitrast"]
+    absminsuit = options["absminsuit"]
+    radius = int(options["radius"])
+    if (radius % 2) == 0:
+        gs.error("Radius should be an odd positive number")
+    focalstat = options["focalstat"]
+    minsuit = float(options["minsuit"])
+    minfrag = float(options["minfrag"])
+    maxgap = float(options["maxgap"])
+    suitreg = options["suitreg"]
+
+    # Flags
+    flag_c = flags["c"]
+    if flag_c:
+        c = "c"
+    else:
+        c = ""
+    flag_d = flags["d"]
+    if flag_d:
+        d = "d"
+    else:
+        d = ""
+    flag_z = flags["z"]
+    flag_k = flags["k"]
+    flag_f = flags["f"]
+    flag_a = flags["a"]
+
+    # Temporary layer names
+    tmp01 = tmpname("tmp01")
+    tmp02 = tmpname("tmp02")
+    tmp02a = tmpname("tmp02a")
+    tmp03 = tmpname("tmp03")
+    tmp04 = tmpname("tmp04")
+    tmp05 = tmpname("tmp05")
+    tmp06 = tmpname("tmp06")
+    tmp07a = tmpname("tmp07a")
+    tmp07 = tmpname("tmp07")
+    tmp08 = tmpname("tmp08")
+    xtralayer = "{}_filledgaps".format(suitreg)
+
+    # Compute neighborhood statistic
+    if (radius > 1) & (len(absminsuit) == 0):
+        gs.message("Computing neighborhood statistic")
+        gs.run_command(
+            "r.neighbors",
+            flags=c,
+            input=suitrast,
+            quiet=True,
+            output=tmp02a,
+            method=focalstat,
+            size=radius,
+        )
+        gs.run_command(
+            "r.series", input=[suitrast, tmp02a], method="maximum", output=tmp02
+        )
+    elif radius > 1:
+        gs.message("Computing neighborhood statistic")
+        try:
+            float(absminsuit)
+        except:
+            gs.error("absminsuit must by numeric or left empty")
+        gs.run_command(
+            "r.neighbors",
+            flags=c,
+            input=suitrast,
+            quiet=True,
+            output=tmp01,
+            method=focalstat,
+            size=radius,
+        )
+        gs.run_command(
+            "r.series", input=[suitrast, tmp01], method="maximum", output=tmp02a
+        )
+        gs.run_command(
+            "r.mapcalc",
+            quiet=True,
+            expression=(
+                "{0} = if({1} > {2},{3},null())".format(
+                    tmp02, suitrast, absminsuit, tmp02a
+                )
+            ),
+        )
+    else:
+        gs.run_command("g.copy", raster=[suitrast, tmp02], quiet=True)
+
+    # Convert suitability to boolean: suitable (1) or not (nodata)
+    gs.message("Creating boolean map suitable/none-suitable")
```

Comment:
Some of there might be `verbose` (messages) rather than `message`.

Reply:
This is intended to give some feedback on progress. Admittedly, not a great way. Ideally, there would be some way of passing on the progress bar of the individual functions used, or to create a progress bar. Any ideas on how to do that, or other clever ways to provide feedback on progress?